### ISSUE TITLE
Query methods are failing due to legacy Array checking methods not working with cross-realm objects

### ISSUE THREAD

**Issue by jmicmoore** [vitaly-t/pg-promise#848]
Hi @vitaly-t, have been using your library for several years now and really enjoying it!  This may be a challenge for me to explain correctly, so please bear with me.  In a nutshell, I ran across an error using pg-promise in a new, very simple project that I believe is happening because of an outdated method of checking to see if a value is an Array, which is not working with cross-realm objects.

### Expected Behavior
The following code should return the selected row from the table
```nodejs
const sql = 'SELECT * FROM naming.sequence_name_info where lower(sequence_type) = $1'
return await getDb().oneOrNone(sql, [sequenceType.toLowerCase()])
```
### Actual Behavior

Instead we get the following error:  ```error:  there is no parameter $1```

### Steps to Reproduce

This is the tricky part.  I don't know the exact combination of libraries that's causing other Array objects to be included in the current realm.  However I can tell you that this affects all of the query methods (none, one, oneOrNone, many, manyOrNone,  any) and it only seems to be an issue when using index variables as parameters (because the array parameter argument is the involved).

And after some debugging, I can pinpoint the exact area of code it's affecting: line 294 of Pg-promise/lib/formatting.js

In the following code snippet, the check to see if values is an instance of Array is returning false when it should be returning true.  This causes the code path to skip formatAs.array(...) and incorrectly go into formatAs.object(...), resulting in the "there is no parameter $1" error.

```node
... // (formatQuery() method
if (values instanceof Array) { // returns false here - should be true!
    // $1, $2,... formatting to be applied;
    return formatAs.array({query, array: values, raw, options}); // it causes the intended method to be skipped...
}
if (!(values instanceof Date || values instanceof Buffer)) {
    // $*propName* formatting to be applied;
    return formatAs.object({query, obj: values, raw, options});  // and this one gets called instead
}
...
```

These screenshots will show that the additional "Array" type(s) enter the current realm *after* debugging has stepped into the oneOrNone() method.

Here you can see the debugger just before calling oneOrNone(); notice in the grey that both ways of checking (isArray1, isArray2) are returning true
 
<img width="404" alt="image" src="https://user-images.githubusercontent.com/12126129/196178238-bafb8c71-8f61-4c35-bca2-c2ae44302179.png">

Here you can see the debugger just inside the oneOrNone() method; I used IntelliJ's watch to evaluate ```values instanceof Array``` and it's returning false.

<img width="371" alt="image" src="https://user-images.githubusercontent.com/12126129/196178864-05c9ed36-cdfc-4fc2-8496-59b8560fc633.png">

### Fix

The MDN [recommendation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#instanceof_vs_array.isarray) is to use ```Array.isArray()``` instead of ```instanceof``` to work for cross-realm objects.
This article goes into a bit more [depth](https://jakearchibald.com/2017/arrays-symbols-realms/)
 

### Environment
- Version of pg-promise: 10.12.0
- OS type (Linux/Windows/Mac): Mac
- Version of Node.js: 14.17.3

**Comment by vitaly-t** [comment]
Fixed, and released v10.12.1 with the fix :wink:

**Comment by jmicmoore** [comment]
Works like a charm - thanks for your quick help with this!  You rock!
